### ISSUE TITLE
Allow Boolean variables to be used in function parameters

### ISSUE THREAD

**Issue by tristanbob** [4ian/GDevelop#2565]
## Description

Almost all function parameters allow a variable to be used to represent the value.  However, this is not true for boolean (yes/no, true/false) parameters.   These parameters only offer the ability to statically pick one of the two values:

![image](https://user-images.githubusercontent.com/8879811/115150550-1b2bf880-a026-11eb-86fd-c0b421d450ef.png)

## Solution suggested

Boolean parameters should allow users to input an expression that can be used to specify a boolean variable.  I think the UI might work similar to Object selection, in that you can type the expression on the top part, but there are easy selections below that line.  

![image](https://user-images.githubusercontent.com/8879811/115150666-a3aa9900-a026-11eb-9312-17fcaf45a120.png)

Propose mockup:

![Screenshot 2021-04-18 091502](https://user-images.githubusercontent.com/8879811/115150856-67c40380-a027-11eb-9ad4-93f2cac6c50a.jpg)

**Comment by 4ian** [comment]
I think there is a card on the roadmap about adding support for Boolean expressions.

Out of curiosity: do you have a specific case in mind where this would be useful?

**Comment by tristanbob** [comment]
In my example game for the OrbitingObjects extension (first screenshot above), I connect all parameters of the extension to UI elements (sliders, checkboxes, toggles).

Without the option to specify a boolean expression, I had to copy two versions of the action, one for each boolean state.  

Does that make sense?

**Comment by tristanbob** [comment]
Another example of what users have to do if they want to use a function with a conditional boolean (in this case, two booleans):

![image](https://user-images.githubusercontent.com/8879811/132105203-0b9835d3-609b-4646-b35b-023c2eeaa43c.png)

**Comment by 4ian** [comment]
Yep that's a real pain.. This is linked to this card: https://trello.com/c/cP4MOzwI/469-add-support-for-expression-returning-booleans
